File: commands/general/ping.py
# https://github.com/viktorexe/terristats-discord-bot
import discord
from discord.ext import commands
from discord import app_commands

class Ping(commands.Cog):
    def __init__(self, bot):
        self.bot = bot

    @app_commands.command(name='ping', description='Check bot latency')
    async def ping(self, interaction: discord.Interaction):
        latency = round(self.bot.latency * 1000)
        embed = discord.Embed(title="🏓 Pong!", color=0x00ff00)
        embed.add_field(name="Latency", value=f"{latency}ms", inline=True)
        await interaction.response.send_message(embed=embed)

async def setup(bot):
    await bot.add_cog(Ping(bot))

File: commands/general/ping.js
// https://github.com/viktorexe/terristats-discord-bot
const { SlashCommandBuilder, EmbedBuilder } = require('discord.js');

module.exports = {
  data: new SlashCommandBuilder()
    .setName('ping')
    .setDescription('Check bot latency'),

  async execute(interaction) {
    // client.ws.ping is the websocket heartbeat latency in milliseconds.
    const latency = Math.round(interaction.client.ws.ping);
    const embed = new EmbedBuilder()
      .setTitle('🏓 Pong!')
      .setColor(0x00ff00)
      .addFields({ name: 'Latency', value: `${latency}ms`, inline: true });
    await interaction.reply({ embeds: [embed] });
  },
};
